core-common/src/main/java/org/mapstruct/Mapper.java

```diff
@@ -148,4 +148,12 @@
      * specified with {@link #config()}.
      */
     MappingInheritanceStrategy mappingInheritanceStrategy() default MappingInheritanceStrategy.EXPLICIT;
+
+    /**
+     * Decide whether we should check null or hasX method before mapping.
+     * It is global, can only override the one on {@link Mapping} if it is UNDEFINED.
+     *
+     * @return strategy about how to do null or has value check
+     */
+    ValueSetCheckStrategy valueSetCheckStrategy() default ValueSetCheckStrategy.IS_NULL_INLINE;
```

Comment:
I am still trying to find a better name. How about `SourceValuePresenceCheckStrategy`? Not exactly the shortest, but maybe better to reason about?